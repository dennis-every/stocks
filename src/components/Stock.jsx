import PropTypes from 'prop-types';
import './Stock.scss';
import { BsBarChartLineFill } from 'react-icons/bs';
import { AiOutlineArrowRight } from 'react-icons/ai';

const Stock = (props) => {
  const {
    symbol,
    companyName,
    sector,
    marketCap,
    beta,
    price,
    exchangeShortName,
    volume,
    lastAnnualDividend,
  } = props;

  return (
    <section>
      <header>
        <div className="icon"><BsBarChartLineFill /></div>
        <h1 className="name">{companyName}</h1>
      </header>
      <div className="breakdown">
        <span>data breakdown</span>
      </div>
      <ul>
        <li className="item even">
          <span>Symbol</span>
          <span>
            {symbol}
            <span className="arrow"><AiOutlineArrowRight /></span>
          </span>
        </li>
        <li className="item odd">
          <span>Market cap</span>
          <span>
            {marketCap.toLocaleString('en-US', {
              style: 'currency',
              currency: 'USD',
              minimumFractionDigits: 0,
              maximumFractionDigits: 0,
            })}
            <span className="arrow"><AiOutlineArrowRight /></span>
          </span>
        </li>
        <li className="item even">
          <span>Sector</span>
          <span>
            {sector}
            <span className="arrow"><AiOutlineArrowRight /></span>
          </span>
        </li>
        <li className="item odd">
          <span>Beta</span>
          <span>
            {beta.toLocaleString('en-US', {
              minimumFractionDigits: 2,
              maximumFractionDigits: 2,
            })}
            <span className="arrow"><AiOutlineArrowRight /></span>
          </span>
        </li>
        <li className="item even">
          <span>Price</span>
          <span>
            {price.toLocaleString('en-US', {
              style: 'currency',
              currency: 'USD',
              minimumFractionDigits: 2,
              maximumFractionDigits: 2,
            })}
            <span className="arrow"><AiOutlineArrowRight /></span>
          </span>
        </li>
        <li className="item odd">
          <span>Exchange</span>
          <span>
            {exchangeShortName}
            <span className="arrow"><AiOutlineArrowRight /></span>
          </span>
        </li>
        <li className="item even">
          <span>Volume</span>
          <span>
            {volume.toLocaleString('en-US', {
              minimumFractionDigits: 0,
              maximumFractionDigits: 0,
            })}
            <span className="arrow"><AiOutlineArrowRight /></span>
          </span>
        </li>
        <li className="item odd">
          <span>Last annual dividend</span>
          <span>
            {lastAnnualDividend.toLocaleString('en-US', {
              style: 'currency',
              currency: 'USD',
              minimumFractionDigits: 2,
              maximumFractionDigits: 2,
            })}
            <span className="arrow"><AiOutlineArrowRight /></span>
          </span>
        </li>
      </ul>
    </section>
  );
};

Stock.propTypes = {
  symbol: PropTypes.string.isRequired,
  companyName: PropTypes.string.isRequired,
  marketCap: PropTypes.number.isRequired,
  sector: PropTypes.string.isRequired,
  beta: PropTypes.number.isRequired,
  price: PropTypes.number.isRequired,
  lastAnnualDividend: PropTypes.number.isRequired,
  volume: PropTypes.number.isRequired,
  exchangeShortName: PropTypes.string.isRequired,
};

export default Stock;
