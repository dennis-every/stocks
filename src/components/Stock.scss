section {
  font-weight: bold;

  header {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    background: #5484e1;
    height: 10rem;

    .icon {
      display: flex;
      justify-content: center;
      text-align: center;
      align-items: center;
      font-size: 5rem;
      color: #2d4472;
    }

    .name {
      display: flex;
      justify-content: center;
      text-align: center;
      align-items: center;
      margin: 0;
    }
  }

  .breakdown {
    text-transform: uppercase;
    font-size: 1rem;
    font-weight: bold;
    padding: 0.5rem 1rem;
    background: #3d5a99;
  }

  ul {
    list-style: none;
    padding: 0;
    margin: 0;

    .item {
      height: 10rem;
      width: 100%;
      display: flex;
      justify-content: space-between;
      align-items: center;

      span {
        text-transform: uppercase;
        padding-left: 1rem;
        padding-right: 1rem;

        .arrow {
          border: 1px solid white;
          border-radius: 50%;
          padding: 0.1rem;
          margin-left: 1rem;
        }
      }
    }

    .even {
      background: #284e97;
    }

    .odd {
      background: #2b55a3;
    }
  }
}
