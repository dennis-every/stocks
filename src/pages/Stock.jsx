import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import StockComponent from '../components/Stock';

const Stock = () => {
  const { slug } = useParams();
  const stocksArray = useSelector((store) => store.stocks.stocksArray);
  const content = stocksArray.find((stock) => stock.symbol === slug);
  const {
    symbol,
    companyName,
    sector,
    marketCap,
    beta,
    price,
    exchangeShortName,
    volume,
    lastAnnualDividend,
  } = content;

  return (
    <>
      <StockComponent
        symbol={symbol}
        companyName={companyName}
        sector={sector}
        marketCap={marketCap}
        beta={beta}
        price={price}
        exchangeShortName={exchangeShortName}
        volume={volume}
        lastAnnualDividend={lastAnnualDividend}
      />
    </>
  );
};

export default Stock;
