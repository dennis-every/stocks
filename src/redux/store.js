import { configureStore } from '@reduxjs/toolkit';
import stocksReducer from './stocks/stocksSlice';

const store = configureStore({
  reducer: {
    stocks: stocksReducer,
  },
});

export default store;
