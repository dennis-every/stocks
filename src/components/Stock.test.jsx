import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Stock from './Stock';
import store from '../redux/store';

const StockObject = {
  symbol: 'AAPL',
  companyName: 'Apple Inc.',
  marketCap: 1382174560000,
  sector: 'Technology',
  beta: 1.2284990000000000076596506914938800036907196044921875,
  price: 318.8899999999999863575794734060764312744140625,
  lastAnnualDividend: 3.0800000000000000710542735760100185871124267578125,
  volume: 51500795,
  exchangeShortName: 'NASDAQ',
};

describe('test suite for Stock component', () => {
  it('expects to render a snapshot of the Stock component', () => {
    const stockComponent = render(
      <Provider store={store}>
        <Stock
          symbol={StockObject.symbol}
          companyName={StockObject.companyName}
          sector={StockObject.sector}
          marketCap={StockObject.marketCap}
          beta={StockObject.beta}
          price={StockObject.price}
          exchangeShortName={StockObject.exchangeShortName}
          volume={StockObject.volume}
          lastAnnualDividend={StockObject.lastAnnualDividend}
        />
      </Provider>,
    );
    expect(stockComponent).toMatchSnapshot();
  });
});
