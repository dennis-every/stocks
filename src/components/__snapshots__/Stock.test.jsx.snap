// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test suite for Stock component expects to render a snapshot of the Stock component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h3>
        Name: Apple Inc.
      </h3>
      <ul>
        <li>
          Symbol: AAPL
        </li>
        <li>
          Market cap: 1382174560000
        </li>
        <li>
          Sector: Technology
        </li>
        <li>
          Beta: 1.228499
        </li>
        <li>
          Price: 318.89
        </li>
        <li>
          Exchange: NASDAQ
        </li>
        <li>
          Volume: 51500795
        </li>
        <li>
          Last annual dividend: 3.08
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <h3>
      Name: Apple Inc.
    </h3>
    <ul>
      <li>
        Symbol: AAPL
      </li>
      <li>
        Market cap: 1382174560000
      </li>
      <li>
        Sector: Technology
      </li>
      <li>
        Beta: 1.228499
      </li>
      <li>
        Price: 318.89
      </li>
      <li>
        Exchange: NASDAQ
      </li>
      <li>
        Volume: 51500795
      </li>
      <li>
        Last annual dividend: 3.08
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
